crates/core/src/subscription/subscription.rs

```diff
@@ -48,6 +50,15 @@ impl Subscription {
     }
 }
 
+// If a RelValue has an id (DataKey) return it directly, otherwise we must construct it from the
+// row itself which can be an expensive operation.
+fn pk_for_row(row: &RelValue) -> PrimaryKey {
```

Comment:
We should really rename `PrimaryKey` down the road. It never was a `PrimaryKey` as much as it was a `RowId`